src/client/mod.rs

```diff
@@ -189,6 +193,10 @@ impl IdGenerator {
     }
 }
 
+impl Default for IdGenerator {
```

Comment:
Oh, BTW, looks like Clippy actually suggests having `Default` derived for this. Any particular reason you didn't do so here?

Reply:
AFAIK deriving `Default` means
```rust
impl Default for Type {
    fn default() -> Self {
        Self {
            field_1: FieldType1::default(), 
            // and so on
        }
    }
}
```
It doesn't call the constructor, since that concept is sadly only conventional. It appears we can depend on that derived `default` methods would return the same values as your current constructors though. In that case, we can redefine your constructors in terms of the `default` methods.

Reply:
Ah, makes sense. Yes, I think it would be nice if we `derive(Default)` and then call `Self::default()` in `new()` (effectively reversing the dependency).

Reply:
@djc: You can merge the existing branch, then I'll make the changes to `imap-proto` required for deriving `Default` in this crate.